include/eve/detail/function/simd/common/broadcast.hpp

```diff
@@ -0,0 +1,42 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2020 Joel FALCOU
+  Copyright 2020 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/pattern.hpp>
+#include <eve/forward.hpp>
+
+namespace eve::detail
+{
+  template<simd_value Wide>
+  EVE_FORCEINLINE auto broadcast_(EVE_SUPPORTS(cpu_), Wide v, auto Index) noexcept
+  {
+    return v[broadcast_n<Index,cardinal_v<Wide>>];
+  }
+
+  template<typename Wide, shuffle_pattern Pattern>
+  EVE_FORCEINLINE auto broadcast_(EVE_SUPPORTS(cpu_), logical<Wide> v, Pattern p) noexcept
+  {
+    constexpr auto sz = Pattern::size(cardinal_v<logical<Wide>>);
+    using   that_t    = as_wide_t<logical<Wide>,fixed<sz>>;
+
+    return that_t{ v.get(p(0,sz)) };
```

Comment:
What's pattern of 0/size? This is not an overload that makes sense

Reply:
p(0,sz) is computing the value of the pattern at index 0 for a given size

Reply:
Fixed